### PR TITLE
fix(labels): update experience for display labels and create labels

### PR DESCRIPTION
**Pull request by vikram-raj**
- Sort labels in alphabetical order in list, detail and quick preview.
- Show random color instead of showing the same color every time while creating the label.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2261-1 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2261-fabric8-planner.badger.fabric8.io

File: src/app/components/work-item-detail/work-item-detail.component.ts [hunk 1]
```diff
@@ -367,6 +367,13 @@ export class WorkItemDetailComponent implements OnInit, OnDestroy {
           this.workItem.relationships.labels.data.map(label => {
             return this.labels.find(l => l.id === label.id);
           });
+          // Sort labels in alphabetical order
+          this.workItem.relationships.labels.data =
+          this.workItem.relationships.labels.data.sort(function(labelA, labelB) {
+            let labelAName = labelA.attributes.name.toUpperCase();
+            let labelBName = labelB.attributes.name.toUpperCase();
+            return (labelAName < labelBName) ? -1 : (labelAName > labelBName) ? 1 : 0;
```
Comment (jarifibrahim) [inline src/app/components/work-item-detail/work-item-detail.component.ts]: You could also do a `return labelAName.localCompare(labelBName)`

Comment (vikram-raj) [reply src/app/components/work-item-detail/work-item-detail.component.ts]: Yes, I will use this. Thanks @jarifibrahim

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2261-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2261-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@vikram-raj fabric8/fabric8-ui:SNAPSHOT-PR-2261-4 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2261-fabric8-planner.badger.fabric8.io

**Comment by nimishamukherjee** [comment]
@rgarg1 @jarifibrahim this PR fixes https://openshift.io/openshiftio/openshiftio/plan/detail/1577

**Comment by rgarg1** [comment]
@vikram-raj @nimishamukherjee The label are now displayed in sorted order. The issue is fixed. 

Note: When the labels start with Uppercase and Lowercase of a same letter, e.g. 'a' & 'A', the sorting is random.

**Comment by rgarg1** [comment]
The label colors are picked randomly too.

**Pull request merged**